llama.cpp

```diff
@@ -641,31 +710,43 @@ static bool llama_model_load(
 
     // load weights
     {
-        size_t total_size = 0;
+        int64_t total_size = 0;
         model.n_loaded = 0;
 
         while (true) {
             int32_t n_dims;
             int32_t length;
             int32_t ftype;
 
-            fin.read(reinterpret_cast<char *>(&n_dims), sizeof(n_dims));
-            fin.read(reinterpret_cast<char *>(&length), sizeof(length));
-            fin.read(reinterpret_cast<char *>(&ftype),  sizeof(ftype));
-
-            if (fin.eof()) {
-                break;
-            }
+            fin.read((char *)&n_dims, 4);
+            if (fin.eof()) break;
```

Comment:
as I mentioned earlier, please check the size intially. check eof is not good idea for every io.